packages/puppeteer-core/src/common/Input.ts

```diff
@@ -438,90 +519,141 @@ export class Mouse {
     this.#keyboard = keyboard;
   }
 
-  /**
-   * Dispatches a `mousemove` event.
-   * @param x - Horizontal position of the mouse.
-   * @param y - Vertical position of the mouse.
-   * @param options - Optional object. If specified, the `steps` property
-   * sends intermediate `mousemove` events when set to `1` (default).
-   */
-  async move(
-    x: number,
-    y: number,
-    options: {steps?: number} = {}
+  async #withRollback(
```

Comment:
So is my understanding correct for the following case?

Two events are dispatched in one task: mousedown, mousemove. Then events happens in the following order:

```
mousedown
mousemove

1) mousdown resolves
2) previous state = current state =  (mousedown + mousemove) at line 531
3) mousemove rejects
4) state = previous state = mousedown + mousemove at line 533
```

So it looks like rollback state is not correct since it has to only include the mousedown state.

Reply:
Fixed.